core/src/test/java/org/elasticsearch/action/admin/indices/create/CreateIndexRequestTests.java

```diff
@@ -113,6 +135,18 @@ private void assertMappingsEqual(Map<String, String> expected, Map<String, Strin
         }
     }
 
+    private void assertAliasesEqual(Set<Alias> expected, Set<Alias> actual) throws IOException {
+        assertEquals(expected, actual);
+
+        for (Alias expectedAlias : expected) {
+            for (Alias actualAlias : actual) {
+                if (expectedAlias.equals(actualAlias)) {
```

Comment:
I guess we will never get here if they are not equal?

Reply:
Yeah; we check above that the two sets are equal; but `Alias.equals` returns true if they have the same name. So at this stage, we check that "equal" aliases have the same parameters, as well.

Maybe we should also check routing params, here?

Reply:
I see why you do that, it would be good to also check routing. It is odd that `Alias#equals` only looks at the name, not sure why but I wouldn't change that at the moment. I just meant that the if is redundant, isn't the equality check already part of the above assertion, which will make the test fail in case it's not satisfied?

Reply:
Ah, no; here, we're iterating over the two sets, but we only want to compare matching aliases. Without the if, we'd be comparing each alias from `expected` to every alias from `actual` - which would of course fail.

Reply:
right! silly me ;) mind leaving a comment saying that we do this because Alias#equals only looks at name?

Reply:
Done 🙂